Implement PRD Generator: AI-assisted Product Requirements Document creation workflow

Zyra-V23 opened the pull request:
# PR Title
"Implement PRD Generator: AI-assisted Product Requirements Document creation workflow"

# PR Notes

## 🚀 Overview

This PR introduces a complete PRD (Product Requirements Document) generation workflow powered by AI models like Claude and Perplexity. The system transforms basic product ideas into comprehensive, professional requirement documents through a guided, multi-stage process that simulates expert input and discussion.

## 🧩 Key Components

### New Core Services
- **Ideation Service**: Generates structured product concepts from simple ideas
- **Discussion Service**: Simulates round-table discussions between expert roles
- **PRD Service**: Creates final PRD documents with customizable formatting

### New Data Models
- `Idea`: Represents initial product concepts with metadata
- `Discussion`: Captures expert discussions and key insights
- `PRD`: Structures the final document with proper versioning

### Workflow Integration
- All commands can be chained for a seamless document creation process
- Interactive user prompts guide users through each step
- Consistent file management in the `prd/` directory

## 🔄 PRD Generation Workflow

The complete workflow consists of four main commands:

1. **Ideation**
   ```bash
   node scripts/dev.js ideate --idea="Product idea description" [--research]
   ```
   Transforms an initial idea into a structured concept document.

2. **Round Table**
   ```bash
   node scripts/dev.js round-table --concept-file=prd/concept.txt [--participants="..."]
   ```
   Simulates a discussion between expert roles about the concept.

3. **Concept Refinement**
   ```bash
   node scripts/dev.js refine-concept --concept-file=prd/concept.txt [--discussion-file=prd/discussion.txt]
   ```
   Improves the concept using insights from the expert discussion.

4. **PRD Generation**
   ```bash
   node scripts/dev.js generate-prd --concept-file=prd/concept_refined.txt [--template=...] [--format=...]
   ```
   Creates the final PRD document based on the refined concept.

## 💡 UI Improvements

- Added `ora` for loading indicators during AI operations
- Enhanced user feedback with color-coded status messages
- Interactive prompts at each step to guide workflow
- Clear next-step suggestions after each command

## 🧪 Testing Status

- 257 total tests (increase from ~126 original tests)
- 204 tests passing
- 49 tests skipped (primarily AI-dependent tests)
- 4 tests failing in commands.test.js (CONFIG.projectVersion issues)

## 📋 Cursor Rules Updates

Updated `.cursor/rules/` files with documentation about:
- New architecture components
- PRD generation workflow
- New feature integration
- External dependency justification

## 🔍 Implementation Notes

This feature creates a bridge between initial product ideation and structured task generation, allowing teams to:
1. Start with just a basic product idea
2. Generate professional PRD documentation
3. Parse the PRD into executable tasks
4. Track implementation progress

This completes the full circle of the Task Master CLI, from idea to implementation tracking.

Zyra-V23 reviewed (commented):
_oupsy_

Zyra-V23 reviewed (commented):
Just /ignore the .cursor/rules :'D sorry abouut the language, sometimes it rebels against me

Zyra-V23 commented:
@eyaltoledano this is the good one, rest you can delete, forget the .txt an .cursor etc, was late and didnt payed so much attention sorry sir

eyaltoledano commented:
LMK if you need help with the merge conflicts. We dont want to overwrite the new work that was done.

Zyra-V23 commented:
> LMK if you need help with the merge conflicts. We dont want to overwrite the new work that was done.

AS LONG AS I REMEMBER, there were not merge conflicts, corefunctionality has been uploaded isolated, so it was 100% working when I did PR'd (Well, didnt check for user bugs etc...).

The difference between this 92 issue and last one was just regarding the isolation of the functionalities.

eyaltoledano commented:
That's nearly impossible, as my update touched like 250 files which is p much the whole project lol

i.e. scripts/modules/commands.s and dev.js and the rules

Zyra-V23 commented:
Ok yes, @eyaltoledano now I get what you mean. I was about to go sleep, if it's not done when I'm back, I'll give an eye :)

Zyra-V23 commented:
I think you were talking about if I was having problem during my merge tests (i did a commit bout that and mocked one test lmao)

Crunchyman-ralph commented:
@Zyra-V23  please resolve conflicts, will be reviewing asap

Zyra-V23 commented on scripts/modules/error-handler.js:
yes! we do have one in mcp-server/src/logger.js :) removing it

Zyra-V23 commented:
@Crunchyman-ralph 
I've reviewed the PRD generation services and addressed your feedback by removing the error-handler.js and prd.txt files. The ServiceInterface is a simple base class for the new services rather than an MCP implementation.
One important integration point remains: the services (Ideation, Discussion, PRD) are fully implemented with complete functionality but aren't yet connected to the command-line interface in commands.js.

For the services to be accessible via task-master ideate or similar commands, we need to:

Register the commands in commands.js
Connect them to the service implementations
Add appropriate help text and parameter definitions

Would you like me to add the command registration to commands.js in a follow-up PR, or would you prefer to integrate them differently into the MCP architecture? I was told to just do the logic and leave the mcp connection part to you guys @eyaltoledano , let me know :)

Crunchyman-ralph commented:
Seems like it is failing the format-check and test. Please fix those failing unit tests

Zyra-V23 commented:
It's failing because error-handler was removed, I will upload it soon and make sure it's clear defined its for handling errors related to prd service

Enviado desde Proton Mail para Android

-------- Mensaje original --------
El 15/4/25 18:09, Ralph Khreish  ha escrito:

> Seems like it is failing the format-check and test. Please fix those failing unit tests
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/eyaltoledano/claude-task-master/pull/92#issuecomment-2806664847), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BPXASXXWI3ZAWKH2TX7ICJ32ZUVNHAVCNFSM6AAAAAB2NECSXKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMBWGY3DIOBUG4).
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> Crunchyman-ralph left a comment [(eyaltoledano/claude-task-master#92)](https://github.com/eyaltoledano/claude-task-master/pull/92#issuecomment-2806664847)
>
> Seems like it is failing the format-check and test. Please fix those failing unit tests
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/eyaltoledano/claude-task-master/pull/92#issuecomment-2806664847), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BPXASXXWI3ZAWKH2TX7ICJ32ZUVNHAVCNFSM6AAAAAB2NECSXKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMBWGY3DIOBUG4).
> You are receiving this because you were mentioned.Message ID: ***@***.***>

eyaltoledano commented:
please rebase and solve any merge conflicts with next

elvenking commented:
Hello guys. 

First of all thanks all of you for putting your work into Task Master :)

As this PR adding a crucial element which i am enthusiastically awaiting, i would like to share with you my recent finding in AI PRD generation scene:

[CreateMVP](https://github.com/rohitg00/CreateMVP)

This repo contains interesting approach to PRD generation. Could you imagine some Task-Master <> CreateMVP integration or at least inspiration for future development of your PRD generation workflow?

mreduar commented:
Hi! Any updates on this PR?

Crunchyman-ralph commented:
@mreduar this PR was a long time ago, I think we would want to re-do this with our new findings and approach

Crunchyman-ralph commented:
Won't be doing a PRD generator in the end, transitioning task-master direction into a task-orchestrator rather than a PRD generator. Had to make a choice, thanks @Zyra-V23 for your efforts, you won't be forgotten!

elvenking commented:
@Crunchyman-ralph do you have some recommended approach or set of tools for PRD generation then ? I believe it is still relevant to task-master to a least guide users in some proven direction :)
